src/rust/zh.manhuagui/src/parser.rs

```diff
@@ -0,0 +1,350 @@
+      )
+      .text()
+	  .read();
+	let desc = html.select("#intro-cut").text().read();
+	let image = format!("https://cf.hamreus.com/cpic/b/{}.jpg", manga_id);
+	let url = format!("https://www.manhuagui.com/comic/{}/", manga_id);
+
+	let manga = Manga {
+		id: manga_id,
+		cover: image,
+		title,
+		author: author.clone(),
+		artist: author,
+		description: desc,
+		url,
+		categories: vec![],
+		status: MangaStatus::Ongoing,
+		nsfw: MangaContentRating::Safe,
+		viewer: MangaViewer::Scroll,
+	};
+
+	Ok(manga)
+}
+
+pub fn get_chapter_list(html: Node) -> Result<Vec<Chapter>> {
+	let mut chapters: Vec<Chapter> = Vec::new();
+	let mut index = 1.0;
+
+	let mut div = html.clone();
+	let hidden = html.html().read().contains("__VIEWSTATE");
+	if hidden {
+		let compressed = html.select("#__VIEWSTATE").attr("value").read();
+		let decompressed =
+			String::from_utf16(&decompress_from_base64(compressed.as_str()).unwrap_or_default()).unwrap_or_default();
+		div = Node::new_fragment(decompressed.as_bytes()).unwrap_or(div);
+	}
+
+	for element in div.select(".chapter-list").array() {
+		let chapt_list_div = match element.as_node() {
+			Ok(node) => node,
+			Err(_) => continue,
+		};
+
+		for ul_ref in chapt_list_div.select("ul").array() {
+			let ul = match ul_ref.as_node() {
+				Ok(node) => node,
+				Err(_) => continue,
+			};
+
+			for li_ref in ul.select("li").array() {
+				let elem = match li_ref.as_node() {
+					Ok(node) => node,
+					Err(_) => continue,
+				};
+
+				let url = elem.select("a").attr("href").read();
+				let id = url.clone().replace("/comic/", "").replace(".html", "");
+				let chapter_id = match id.split('/').last() {
+					Some(id) => String::from(id),
+					None => String::new(),
+				};
+				let title = elem.select("a").attr("title").read();
+				let chapter_or_volume = title
+					.clone()
+					.replace(['第', '话', '卷'], "")
+					.parse::<f32>()
+					.unwrap_or(index);
+				let ch = if title.contains('卷') { -1.0 } else { chapter_or_volume };
+				let vo = if title.contains('卷') { chapter_or_volume } else { -1.0 };
+
+				let chapter = Chapter {
+					id: chapter_id,
+					title,
+					volume: vo,
+					chapter: ch,
+					date_updated: -1.0,
+					scanlator: String::new(),
+					url,
+					lang: String::from("zh"),
+				};
+
+				chapters.push(chapter);
+				index += 1.0;
+			}
+		}
+	}
+
+	Ok(chapters)
+}
+
+pub fn get_page_list(base_url: String) -> Result<Vec<Page>> {
+	let mut pages: Vec<Page> = Vec::new();
+
+	let html = Request::new(base_url.as_str(), HttpMethod::Get).html()?;
+
+	let decoder = Decoder::new(html.html().read());
+	let (path, pages_str) = decoder.decode();
+
+	for (index, str) in pages_str.into_iter().enumerate() {
+		let url = format!("https://i.hamreus.com{}{}", path, str);
+		let encoded_url = helper::encode_uri(&url);
+		let page: Page = Page {
+			index: index as i32,
+			url: encoded_url,
+			base64: String::new(),
+			text: String::new(),
+		};
+		pages.push(page);
```

Comment:
str is already encoded as far as I can tell, so attempting to encode it a second time causes it to break. I think only the path needs to be encoded.
```suggestion
		let encoded_path = helper::encode_uri(&path);
		let url = format!("https://i.hamreus.com{}{}", encoded_path, str);
		pages.push(Page {
			index: index as i32,
			url,
			base64: String::new(),
			text: String::new(),
		});
```